src/nvim/diff.c

```diff
@@ -1733,25 +1743,713 @@ void diff_clear(tabpage_T *tp)
+/// description, see "linematch_3buffers" for the current diff hunk "dp",
+/// the "score" and path at index "i","j","k" is updated from the path
+/// from "_i","_j","_k" choice is represented by enum value "choice"
+///
+/// @param dp
+/// @param df_pathmatrix3
+/// @param score
+/// @param i
+/// @param j
+/// @param k
+/// @param _i
+/// @param _j
+/// @param _k
+/// @param choice
+void update_path3(diff_T *dp, diffcomparisonpath3_T ***df_pathmatrix3,
+                  int score, int i, int j, int k, int _i, int _j, int _k,
+                  enum path3_choice choice)
+{
+  df_pathmatrix3[i][j][k].df_lev_score = score;
+  for (int __k = 0; __k <= df_pathmatrix3[_i][_j][_k].path_index; __k++) {
+    df_pathmatrix3[i][j][k].df_path3[__k]=
+      df_pathmatrix3[_i][_j][_k].df_path3[__k]; }
+  df_pathmatrix3[i][j][k].path_index = df_pathmatrix3[_i][_j][_k].path_index;
+  df_pathmatrix3[i][j][k].df_path3
+    [df_pathmatrix3[i][j][k].path_index] = choice;  // this choice
+  df_pathmatrix3[i][j][k].path_index++;
+}
+/// helper function for the diff alignment algorithm copy the newly found
+/// optimal path to an index in the tensor for more detailed algorith
+/// description, see "linematch_2buffers" for the current diff hunk "dp",
+/// the "score" and path at index "i","j"is updated from the path from
+/// "_i","_j",choice is represented by enum value "choice"
+///
+/// @param dp
+/// @param df_pathmatrix2
+/// @param score
+/// @param i
+/// @param j
+/// @param _i
+/// @param _j
+/// @param choice
+void update_path2(diff_T *dp, diffcomparisonpath2_T **df_pathmatrix2,
+                  int score, int i, int j, int _i, int _j,
+                  enum path2_choice choice)
+{
+  df_pathmatrix2[i][j].df_lev_score = score;
+  for (int k = 0; k <= df_pathmatrix2[_i][_j].path_index; k++) {
+    df_pathmatrix2[i][j].df_path2[k]=
+      df_pathmatrix2[_i][_j].df_path2[k]; }
+  df_pathmatrix2[i][j].path_index = df_pathmatrix2[_i][_j].path_index;
+  df_pathmatrix2[i][j].df_path2
+    [df_pathmatrix2[i][j].path_index] = choice;  // this choice
+  df_pathmatrix2[i][j].path_index++;
+}
+/// helper function for the calculation of the levenshtein distance
+///
+/// @param s1
+/// @param s2
+long levmin(long a, long b, long c)
+{
+        if (a <= b && a <= c) {
+                return a;
+        } else if (b <= a && b <= c) {
+                return b;
+        } else if (c <= a && c <= b) {
+                return c;
+        }
+        return 0;
+}
+/// levenshtein distance calculation. Return the calculated levenshtein distance
+/// between string "s1" and "s2".
+///
+/// @param s1
+/// @param s2
+long levenshtein(const char_u *s1, const char_u *s2)
+{
+    long x, y;
+    long s1len=(long)STRLEN(s1), s2len=(long)STRLEN(s2);
+    unsigned long **matrix = xmalloc(sizeof( long * ) * (s2len+1));
+    for (long i = 0; i < (s2len+1); i++) {
+        matrix[i]=xmalloc(sizeof(long) * (s1len+1));
+    }
+
+    matrix[0][0] = 0;
```

Comment:
Did the linter really skip this function? The indentation seems a bit off.

Reply:
I've double checked it by running "make lint", it doesn't show any errors there. I agree it looks strange, I've noticed also some areas where I think it probably should be showing warnings where I forget to put spaces around equals signs for example if you look at the beginning of the function I wrote called "count_matched_chars". I might be using the linter incorrectly

Reply:
I just went through and tried to fix all the white space related problems that I've noticed

Reply:
There you go, someone turned on CI checks. There are linter complaints there.

Reply:
Are all the formatting issues fixed now? I saw in one of the failed tests the command run was "make clint-full" I fixed all the errors output from that, there were more than what I was previously seeing with "make lint".
Right now when I run "make clint-full" I get some errors in src/nvim/funcs.c and src/ex_docmd.c, but these are far away from any code that I modified.